packages/@aws-cdk/aws-batch/lib/job-queue.ts

```diff
@@ -0,0 +1,139 @@
+import { Construct, IResource, Resource, Stack } from '@aws-cdk/core';
+import { CfnJobQueue } from './batch.generated';
+import { ComputeEnvironment, IComputeEnvironment } from './compute-environment';
+
+/**
+ * Properties for mapping a compute environment to a job queue
+ */
+export interface JobQueueComputeEnvironment {
+  /**
+   * The batch compute environment to use for processing submitted jobs to this queue
+   */
+  readonly computeEnvironment: IComputeEnvironment;
+
+  /**
+   * The order in which this compute environment will be selected for dynamic allocation of resources to process submitted jobs
+   */
+  readonly order: number;
+}
+
+/**
+ * Properties of a batch job queue
+ */
+export interface JobQueueProps {
+  /**
+   * A name for the job queue.
+   *
+   * Up to 128 letters (uppercase and lowercase), numbers, hyphens, and underscores are allowed.
+   *
+   * @default Cloudformation-generated name
+   */
+  readonly jobQueueName?: string;
+
+  /**
+   * The set of compute environments mapped to a job queue and their order relative to each other. The job scheduler uses this parameter to
+   * determine which compute environment should execute a given job. Compute environments must be in the VALID state before you can associate them
+   * with a job queue. You can associate up to three compute environments with a job queue.
+   *
+   * @default Default-Compute-Environment
+   */
+  readonly computeEnvironmentOrder?: JobQueueComputeEnvironment[];
```

Comment:
Why the name `computeEnvironmentOrder`? Why not `computeEnvironments`?

Reply:
Hm, not sure. But its better your way. Best to make it easier to understand vs. matching CF templates directly.